lib/pbench/server/api/__init__.py

```diff
@@ -75,6 +79,103 @@ def create_app():
     return app
 
 
+def _build_cors_preflight_response():
+    response = make_response()
+    response.headers.add("Access-Control-Allow-Origin", "*")
+    response.headers.add("Access-Control-Allow-Headers", "*")
+    response.headers.add("Access-Control-Allow-Methods", "*")
+    return response
+
+
+def _corsify_actual_response(response):
+    response.headers.add("Access-Control-Allow-Origin", "*")
+    return response
+
+
+class GraphQL(Resource):
+    def post(self):
+        global app
+        graphQL = app.config_graphql.get("server")
+        json_data = request.get_json()
+
+        if not json_data:
+            app.logger.debug("Invalid query object")
+            abort(400, message="Invalid query object in request")
+
+        try:
+            # query GraphQL
+            response = requests.post(graphQL, json=json_data)
+
+            # construct response object
+            response = make_response(response.text)
+            response = _corsify_actual_response(response)
+        except Exception:
+            app.logger.exception("There was something wrong with the POST request.")
+            abort(500, message="There was something wrong with your graphql request")
+        response.status_code = 200
+        return response
+
+    # From the client side, an OPTIONS request is initiated before every POST to request CORS.
+    # On the server side, the server must specify the response headers to grant CORS accordingly.
+
+    def options(self):
+        try:
+            response = _build_cors_preflight_response()
+        except Exception:
+            app.logger.exception("There was something wrong with the OPTIONS request.")
```

Comment:
Can this be decoded better? Seems any exception here is likely to indicate an improbable Flask problem with make_response or with the created response object itself, not the client's request.